diff --git a/avidemux_core/ADM_ffmpeg/libavformat/flvenc.c b/avidemux_core/ADM_ffmpeg/libavformat/flvenc.c
index 709ef07..a573c63 100644
--- a/avidemux_core/ADM_ffmpeg/libavformat/flvenc.c
+++ b/avidemux_core/ADM_ffmpeg/libavformat/flvenc.c
@@ -406,7 +406,8 @@ AVOutputFormat flv_muxer = {
     "video/x-flv",
     "flv",
     sizeof(FLVContext),
-#if CONFIG_LIBMP3LAME
+//MEANX #if CONFIG_LIBMP3LAME
+#ifdef HAVE_LIBMP3LAME
     CODEC_ID_MP3,
 #else // CONFIG_LIBMP3LAME
     CODEC_ID_ADPCM_SWF,
@@ -418,3 +419,9 @@ AVOutputFormat flv_muxer = {
     .codec_tag= (const AVCodecTag* const []){flv_video_codec_ids, flv_audio_codec_ids, 0},
     .flags= AVFMT_GLOBALHEADER | AVFMT_VARIABLE_FPS,
 };
+// MEANX
+int flvenc_init(void)
+{
+    av_register_output_format(&flv_muxer);
+    return 0;
+}
\ No newline at end of file
