diff --git a/avidemux_core/ADM_ffmpeg/libavformat/movenc.c b/avidemux_core/ADM_ffmpeg/libavformat/movenc.c
index b1ad714..daf70fd 100644
--- a/avidemux_core/ADM_ffmpeg/libavformat/movenc.c
+++ b/avidemux_core/ADM_ffmpeg/libavformat/movenc.c
@@ -2299,3 +2299,15 @@ AVOutputFormat ipod_muxer = {
     .codec_tag = (const AVCodecTag* const []){codec_ipod_tags, 0},
 };
 #endif
+/* MEANX */
+int movenc_init(void)
+{
+    av_register_output_format(&mov_muxer);
+    av_register_output_format(&tgp_muxer);
+    av_register_output_format(&mp4_muxer);
+    av_register_output_format(&psp_muxer);
+    av_register_output_format(&tg2_muxer);
+    av_register_output_format(&ipod_muxer);
+    return 0;
+}
+/* MEANX */
