diff --git a/avidemux_core/ADM_ffmpeg/libavformat/mpegenc.c b/avidemux_core/ADM_ffmpeg/libavformat/mpegenc.c
index b37a774..32ce21b 100644
--- a/avidemux_core/ADM_ffmpeg/libavformat/mpegenc.c
+++ b/avidemux_core/ADM_ffmpeg/libavformat/mpegenc.c
@@ -1300,3 +1300,13 @@ AVOutputFormat mpeg2dvd_muxer = {
     mpeg_mux_end,
 };
 #endif
+// MEANX
+//
+int mpegenc_init(void)
+{
+    av_register_output_format(&mpeg2dvd_muxer);
+    av_register_output_format(&mpeg2svcd_muxer);
+    av_register_output_format(&mpeg2vob_muxer);
+    av_register_output_format(&mpeg1vcd_muxer);
+    return 0;
+}
\ No newline at end of file
