diff --git a/avidemux_core/ADM_ffmpeg/libavformat/mpegtsenc.c b/avidemux_core/ADM_ffmpeg/libavformat/mpegtsenc.c
index bfd21db..85a6363 100644
--- a/avidemux_core/ADM_ffmpeg/libavformat/mpegtsenc.c
+++ b/avidemux_core/ADM_ffmpeg/libavformat/mpegtsenc.c
@@ -958,3 +958,8 @@ AVOutputFormat mpegts_muxer = {
     mpegts_write_packet,
     mpegts_write_end,
 };
+int mpegtsenc_init(void)
+{
+    av_register_output_format(&mpegts_muxer);
+    return 0;
+}
\ No newline at end of file
