diff --git a/avidemux_core/ADM_ffmpeg/libavformat/utils.c b/avidemux_core/ADM_ffmpeg/libavformat/utils.c
index c8e6d66..ac532d4 100644
--- a/avidemux_core/ADM_ffmpeg/libavformat/utils.c
+++ b/avidemux_core/ADM_ffmpeg/libavformat/utils.c
@@ -50,13 +50,14 @@ unsigned avformat_version(void)
 
 const char *avformat_configuration(void)
 {
-    return FFMPEG_CONFIGURATION;
+    return "avidemux"; // MEANX FFMPEG_CONFIGURATION;
 }
 
 const char *avformat_license(void)
 {
 #define LICENSE_PREFIX "libavformat license: "
-    return LICENSE_PREFIX FFMPEG_LICENSE + sizeof(LICENSE_PREFIX) - 1;
+        return LICENSE_PREFIX "GPL";
+//MEANX    return LICENSE_PREFIX FFMPEG_LICENSE + sizeof(LICENSE_PREFIX) - 1;
 }
 
 /* fraction handling */
