--- build/make/configure.sh.orig	2010-10-28 14:14:14 +0100
+++ build/make/configure.sh	2010-11-27 11:45:42 +0000
@@ -877,8 +877,12 @@
         [ "${AS##*/}" = nasm ] && add_asflags -Ox
         AS_SFX=.asm
         case  ${tgt_os} in
-            win*)
-                add_asflags -f win${bits}
+            win32)
+                add_asflags -f win32
+                enabled debug && add_asflags -g dwarf2
+            ;;
+			win64)
+                add_asflags -f x64
                 enabled debug && add_asflags -g dwarf2
             ;;
             linux*|solaris*)
