--- vp8/vp8_common - Copy.mk	2012-01-28 00:06:40 +0530
+++ vp8/vp8_common.mk	2012-01-30 17:49:50 +0530
@@ -68,7 +68,8 @@
 VP8_COMMON_SRCS-yes += common/reconintra4x4.c
 VP8_COMMON_SRCS-yes += common/setupintrarecon.c
 VP8_COMMON_SRCS-yes += common/swapyv12buffer.c
-
+VP8_COMMON_SRCS-yes += encoder/variance_c.c
+VP8_COMMON_SRCS-yes += encoder/sad_c.c
 
 
 VP8_COMMON_SRCS-$(CONFIG_POSTPROC_VISUALIZER) += common/textblit.c
