diff --git a/tinypy/builtins.c b/tinypy/builtins.c
index 1919bd8..0511ffb 100644
--- a/tinypy/builtins.c
+++ b/tinypy/builtins.c
@@ -128,7 +128,7 @@ tp_obj tp_float(TP) {
         char s[32]; memset(s,0,v.string.len+1);
         memcpy(s,v.string.val,v.string.len);
         if (strchr(s,'.')) { return tp_number(atof(s)); }
-        return(tp_number(strtol(s,0,ord)));
+        return(tp_number(strtoll(s,0,ord))); /* MEANX */
     }
     tp_raise(tp_None,tp_string("(tp_float) TypeError: ?"));
 }
