--- libavformat/file.c.old	2010-12-20 13:49:28 +0000
+++ libavformat/file.c	2010-12-20 13:49:29 +0000
@@ -30,6 +30,75 @@
 #include <stdlib.h>
 #include "os_support.h"
 
+// GRUNTSTER start
+#ifdef __WIN32
+#include <windows.h>
+
+int utf8StringToWideChar(const char *utf8String, int utf8StringLength, wchar_t *wideCharString);
+int ADM_open(const char *path, int oflag, ...);
+
+int utf8StringToWideChar(const char *utf8String, int utf8StringLength, wchar_t *wideCharString)
+{
+	int wideCharStringLength = MultiByteToWideChar(CP_UTF8, 0, utf8String, utf8StringLength, NULL, 0);
+
+	if (wideCharString)
+		MultiByteToWideChar(CP_UTF8, 0, utf8String, utf8StringLength, wideCharString, wideCharStringLength);
+
+	return wideCharStringLength;
+}
+
+int ADM_open(const char *path, int oflag, ...)
+{
+	int fileNameLength = utf8StringToWideChar(path, -1, NULL);
+	wchar_t wcFile[fileNameLength];
+	int creation = 0, access = 0;
+	HANDLE hFile;
+
+	utf8StringToWideChar(path, -1, wcFile);
+
+	if (oflag & O_WRONLY || oflag & O_RDWR)
+	{
+		access = GENERIC_WRITE;
+
+		if (oflag & O_RDWR)
+			access |= GENERIC_READ;
+
+		if (oflag & O_CREAT)
+		{
+			if (oflag & O_EXCL)
+				creation = CREATE_NEW;
+			else if (oflag & O_TRUNC)
+				creation = CREATE_ALWAYS;
+			else
+				creation = OPEN_ALWAYS;
+		}
+		else if (oflag & O_TRUNC)
+			creation = TRUNCATE_EXISTING;
+	}
+	else if (oflag & O_RDONLY)
+		creation = OPEN_EXISTING;
+
+	if (creation & GENERIC_WRITE)
+	{
+		hFile = CreateFileW(wcFile, access, 0, NULL, creation, 0, NULL);
+
+		if (hFile == INVALID_HANDLE_VALUE)
+			return -1;
+		else
+			CloseHandle(hFile);
+	}
+
+	hFile = CreateFileW(wcFile, access, FILE_SHARE_READ, NULL, creation, 0, NULL);
+
+	if (hFile == INVALID_HANDLE_VALUE)
+		return -1;
+	else
+		return _open_osfhandle((intptr_t)hFile, oflag);
+}
+
+#define open ADM_open
+#endif
+// GRUNTSTER end
 
 /* standard file protocol */
 
