--- libavcodec/vc1dec.c.old	2010-12-20 13:49:28 +0000
+++ libavcodec/vc1dec.c	2010-12-20 13:49:28 +0000
@@ -3146,7 +3146,10 @@
     AVFrame *pict = data;
     uint8_t *buf2 = NULL;
     const uint8_t *buf_start = buf;
-
+/** MEANX **/
+    if(s->flags& CODEC_FLAG_LOW_DELAY)
+        s->low_delay=1;
+/** /MEANX **/
     /* no supplementary picture */
     if (buf_size == 0) {
         /* special case for last picture */
