diff --git a/src/x264.mk b/src/x264.mk
index 0170301b..1189ccd0 100644
--- a/src/x264.mk
+++ b/src/x264.mk
@@ -3,20 +3,13 @@
 PKG             := x264
 $(PKG)_WEBSITE  := https://www.videolan.org/developers/x264.html
 $(PKG)_IGNORE   :=
-$(PKG)_VERSION  := 20180806-2245
-$(PKG)_CHECKSUM := 9f876c88aeb21fa9315e4a078931faf6fc0d3c3f47e05a306d2fdc62ea0afea2
-$(PKG)_SUBDIR   := $(PKG)-snapshot-$($(PKG)_VERSION)
-$(PKG)_FILE     := $(PKG)-snapshot-$($(PKG)_VERSION).tar.bz2
-$(PKG)_URL      := https://download.videolan.org/pub/videolan/$(PKG)/snapshots/$($(PKG)_FILE)
+$(PKG)_VERSION  := 0.159-20200409git296494a
+$(PKG)_CHECKSUM := 3c884571fccd7f5a4e3b4e08c6386ad356f15ea2c381cf464e87a70af263c144
+$(PKG)_SUBDIR   := $(PKG)-$($(PKG)_VERSION)
+$(PKG)_FILE     := $(PKG)-$($(PKG)_VERSION).tar.bz2
+$(PKG)_URL      :=
 $(PKG)_DEPS     := cc liblsmash $(BUILD)~nasm
 
-define $(PKG)_UPDATE
-    $(WGET) -q -O- 'https://git.videolan.org/?p=x264.git;a=shortlog' | \
-    $(SED) -n 's,.*\([0-9]\{4\}\)-\([0-9]\{2\}\)-\([0-9]\{2\}\).*,\1\2\3-2245,p' | \
-    $(SORT) | \
-    tail -1
-endef
-
 define $(PKG)_BUILD
     cd '$(BUILD_DIR)' && AS='$(PREFIX)/$(BUILD)/bin/nasm' '$(SOURCE_DIR)/configure'\
         $(MXE_CONFIGURE_OPTS) \
