diff --git a/avidemux_core/ADM_ffmpeg/libavutil/bswap.h b/avidemux_core/ADM_ffmpeg/libavutil/bswap.h
index 303bcf3..6a129f3 100644
--- a/avidemux_core/ADM_ffmpeg/libavutil/bswap.h
+++ b/avidemux_core/ADM_ffmpeg/libavutil/bswap.h
@@ -27,7 +27,7 @@
 #define AVUTIL_BSWAP_H
 
 #include <stdint.h>
-#include "libavutil/avconfig.h"
+#include "ffmpeg_config/libavutil/avconfig.h" //MEANX
 #include "attributes.h"
 
 #ifdef HAVE_AV_CONFIG_H
