diff --git a/avidemux_core/ADM_ffmpeg/libavutil/common.h b/avidemux_core/ADM_ffmpeg/libavutil/common.h
index 6404076..3fde7df 100644
--- a/avidemux_core/ADM_ffmpeg/libavutil/common.h
+++ b/avidemux_core/ADM_ffmpeg/libavutil/common.h
@@ -35,7 +35,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include "attributes.h"
-#include "libavutil/avconfig.h"
+#include "ffmpeg_config/libavutil/avconfig.h" //MEANX
 
 #if AV_HAVE_BIGENDIAN
 #   define AV_NE(be, le) (be)
