diff --git a/avidemux_core/ADM_ffmpeg/libavutil/intreadwrite.h b/avidemux_core/ADM_ffmpeg/libavutil/intreadwrite.h
index 1849a64..d6e2dcb 100644
--- a/avidemux_core/ADM_ffmpeg/libavutil/intreadwrite.h
+++ b/avidemux_core/ADM_ffmpeg/libavutil/intreadwrite.h
@@ -20,7 +20,7 @@
 #define AVUTIL_INTREADWRITE_H
 
 #include <stdint.h>
-#include "libavutil/avconfig.h"
+#include "ffmpeg_config/libavutil/avconfig.h" //MEANX
 #include "attributes.h"
 #include "bswap.h"
 
