diff --git a/avidemux_core/ADM_ffmpeg/libavutil/pixfmt.h b/avidemux_core/ADM_ffmpeg/libavutil/pixfmt.h
index 8ec91c8..6882d41 100644
--- a/avidemux_core/ADM_ffmpeg/libavutil/pixfmt.h
+++ b/avidemux_core/ADM_ffmpeg/libavutil/pixfmt.h
@@ -29,7 +29,7 @@
  * header, so it should not be directly included in your projects.
  */
 
-#include "libavutil/avconfig.h"
+#include "ffmpeg_config/libavutil/avconfig.h" // MEANX
 
 /**
  * Pixel format. Notes:
