--- /d/dev/adm_2.6_dev/avidemux_core/ffmpeg_package/ffmpeg/./configure	2012-07-26 15:54:32 +1000
+++ ./configure	2012-07-26 15:54:32 +1000
@@ -1173,6 +1173,7 @@
     dlfcn_h
     dlopen
     dos_paths
+	dxva_h
     ebp_available
     ebx_available
     exp2
@@ -3125,6 +3126,7 @@
 check_func_headers glob.h glob
 
 check_header dlfcn.h
+check_header dxva.h
 check_header dxva2api.h -D_WIN32_WINNT=0x0600
 check_header libcrystalhd/libcrystalhd_if.h
 check_header malloc.h
