--- libavformat/matroskaenc.c.orig	2019-09-08 19:44:28.818111882 +0200
+++ libavformat/matroskaenc.c	2019-09-08 19:44:28.821111864 +0200
@@ -2783,7 +2783,7 @@
         if (pkt->size > 2 && (AV_RB16(pkt->data) & 0xfff0) == 0xfff0)
             ret = ff_stream_add_bitstream_filter(st, "aac_adtstoasc", NULL);
     } else if (st->codecpar->codec_id == AV_CODEC_ID_VP9) {
-        ret = ff_stream_add_bitstream_filter(st, "vp9_superframe", NULL);
+        // MEANX ret = ff_stream_add_bitstream_filter(st, "vp9_superframe", NULL);
     }
 
     return ret;
