--- libavformat/matroskaenc.c.orig	2025-03-09 19:40:48.595534686 +0100
+++ libavformat/matroskaenc.c	2025-03-09 19:40:48.597184516 +0100
@@ -3527,7 +3527,7 @@
         if (pkt->size > 2 && (AV_RB16(pkt->data) & 0xfff0) == 0xfff0)
             ret = ff_stream_add_bitstream_filter(st, "aac_adtstoasc", NULL);
     } else if (st->codecpar->codec_id == AV_CODEC_ID_VP9) {
-        ret = ff_stream_add_bitstream_filter(st, "vp9_superframe", NULL);
+        // MEANX ret = ff_stream_add_bitstream_filter(st, "vp9_superframe", NULL);
     } else if (CONFIG_MATROSKA_MUXER &&
                st->codecpar->codec_id == AV_CODEC_ID_HDMV_PGS_SUBTITLE) {
         ret = ff_stream_add_bitstream_filter(st, "pgs_frame_merge", NULL);
