--- libavutil/file_open.c.orig	2018-11-18 01:58:00.000000000 +0100
+++ libavutil/file_open.c	2019-05-05 22:08:47.119935074 +0200
@@ -38,12 +38,15 @@
 #include <share.h>
 #include <errno.h>
 #include "wchar_filename.h"
+#include "avstring.h"
 
 static int win32_open(const char *filename_utf8, int oflag, int pmode)
 {
     int fd;
     wchar_t *filename_w;
 
+    av_strstart(filename_utf8, "//", &filename_utf8); // remove leading double slashes - eum.
+
     /* convert UTF-8 to wide chars */
     if (utf8towchar(filename_utf8, &filename_w))
         return -1;
