--- /c/dev/adm_2.6_dev/avidemux_core/ffmpeg_package/patches/ffmpeg/libavcodec/x86/fmtconvert_mmx.c	2012-04-18 15:14:07 +1000
+++ libavcodec/x86/fmtconvert_mmx.c	2012-04-18 15:14:07 +1000
@@ -127,6 +127,7 @@
                 c->float_to_int16_interleave = float_to_int16_interleave_3dn2;
             }
         }
+#if 0 // MEANX : Disable as we cant guarantee output buffer is aligned
         if (HAVE_SSE && mm_flags & AV_CPU_FLAG_SSE) {
             c->int32_to_float_fmul_scalar = ff_int32_to_float_fmul_scalar_sse;
             c->float_to_int16 = ff_float_to_int16_sse;
@@ -138,6 +139,7 @@
             c->float_to_int16 = ff_float_to_int16_sse2;
             c->float_to_int16_interleave = float_to_int16_interleave_sse2;
         }
+#endif
     }
 #endif
 }
