--- libavcodec/libavcodec.v.org	2012-10-08 09:26:51.041964900 +0200
+++ libavcodec/libavcodec.v	2012-10-08 09:27:25.229965347 +0200
@@ -29,5 +29,10 @@
                 ff_vdpau_vc1_decode_picture;
                 ff_dnxhd_get_cid_table;
                 ff_dnxhd_cid_table;
+                # Avidemux
+                ff_h264_info;
+                ff_golomb_vlc_len;
+                ff_ue_golomb_vlc_code;
+                ff_se_golomb_vlc_code;
         local:  *;
 };
