--- libavformat/mpegtsenc.c.orig	2017-10-15 17:59:38.000000000 +0200
+++ libavformat/mpegtsenc.c	2017-10-20 21:57:43.625220439 +0200
@@ -1434,7 +1434,7 @@
             av_log(s, AV_LOG_ERROR, "H.264 bitstream malformed, "
                    "no startcode found, use the video bitstream filter 'h264_mp4toannexb' to fix it "
                    "('-bsf:v h264_mp4toannexb' option with ffmpeg)\n");
-            return AVERROR_INVALIDDATA;
+            // MEANX : Avidemux does not align the packet well, ignore this error    return AVERROR_INVALIDDATA;
         }
         av_log(s, AV_LOG_WARNING, "H.264 bitstream error, startcode missing, size %d", pkt->size);
         if (pkt->size)
