--- libavcodec/hevc_parse.c.orig	2020-03-29 00:49:17.000000000 +0100
+++ libavcodec/hevc_parser.c	2020-03-29 00:47:31.000000000 +0100
@@ -31,6 +31,13 @@
 #include "internal.h"
 #include "parser.h"
 
+/* MEANX */
+const HEVCSPS *ff_hevc_parser_get_sps(AVCodecParserContext *s);
+const HEVCPPS *ff_hevc_parser_get_pps(AVCodecParserContext *s);
+const HEVCVPS *ff_hevc_parser_get_vps(AVCodecParserContext *s);
+/* /MEANX */
+
+
 #define START_CODE 0x000001 ///< start_code_prefix_one_3bytes
 
 #define IS_IRAP_NAL(nal) (nal->type >= 16 && nal->type <= 23)
@@ -171,6 +178,50 @@
 }
 
 /**
+ \fn ff_hevc_parser_get_sps
+\brief meanx : extract sps from context
+*/
+const HEVCSPS *ff_hevc_parser_get_sps(AVCodecParserContext *s)
+{
+      HEVCParserContext *ctx = s->priv_data;
+      if(!ctx) return NULL;
+      HEVCParamSets     *ps = &ctx->ps;
+      if(!ps) return NULL;
+
+     const HEVCSPS *sps = ps->sps;
+     return sps;
+}
+/**
+ \fn ff_hevc_parser_get_pps
+\brief meanx : extract pps from context
+*/
+const HEVCPPS *ff_hevc_parser_get_pps(AVCodecParserContext *s)
+{
+      HEVCParserContext *ctx = s->priv_data;
+      if(!ctx) return NULL;
+      HEVCParamSets     *ps = &ctx->ps;
+      if(!ps) return NULL;
+
+     const HEVCPPS *pps = ps->pps;
+     return pps;
+}
+/**
+ \fn ff_hevc_parser_get_vps
+\brief meanx : extract vps from context
+*/
+const HEVCVPS *ff_hevc_parser_get_vps(AVCodecParserContext *s)
+{
+      HEVCParserContext *ctx = s->priv_data;
+      if(!ctx) return NULL;
+      HEVCParamSets     *ps = &ctx->ps;
+      if(!ps) return NULL;
+
+     const HEVCVPS *vps = ps->vps;
+     return vps;
+}
+// /MEANX
+
+/**
  * Parse NAL units of found picture and decode some basic information.
  *
  * @param s parser context.
