--- libavcodec/h263dec.c.old	2010-12-20 13:49:21 +0000
+++ libavcodec/h263dec.c	2010-12-20 13:49:21 +0000
@@ -120,6 +120,25 @@
 
     return 0;
 }
+/* MeanX : Ugly patch to detect vo ppacked stuff ... */
+int av_is_voppacked(AVCodecContext *avctx, int *vop_packed, int *gmc, int *qpel);
+
+int av_is_voppacked(AVCodecContext *avctx, int *vop_packed, int *gmc, int *qpel)
+{
+    MpegEncContext *s = avctx->priv_data;
+    // set sane default
+    *vop_packed=0;
+    *gmc=0;
+    *qpel=0;
+    if(avctx->codec->id!=CODEC_ID_MPEG4) return 0;
+    	
+  	*vop_packed=(s->divx_packed);
+	*qpel=s->quarter_sample;
+	*gmc=0;	// FIXME
+	return 1;
+
+  }
+  /* MeanX */
 
 av_cold int ff_h263_decode_end(AVCodecContext *avctx)
 {
@@ -422,6 +441,12 @@
     } else {
         ret = h263_decode_picture_header(s);
     }
+	//MEANX we need to do it here also for quicktime file / ctts atom 
+        // we need the correct frame type, and qt file may contain 
+        // vop not coded frame.
+        pict->pict_type=s->current_picture.pict_type= s->pict_type;
+        pict->key_frame=s->current_picture.key_frame= s->pict_type == FF_I_TYPE;
+        //MEANX
 
     if(ret==FRAME_SKIPPED) return get_consumed_bytes(s, buf_size);
 
@@ -717,6 +742,14 @@
 
 assert(s->current_picture.pict_type == s->current_picture_ptr->pict_type);
 assert(s->current_picture.pict_type == s->pict_type);
+/* MEANX */
+  if(s->current_picture_ptr)
+      s->current_picture_ptr->opaque=pict->opaque;
+/* MEANX */
+
+
+
+
     if (s->pict_type == FF_B_TYPE || s->low_delay) {
         *pict= *(AVFrame*)s->current_picture_ptr;
     } else if (s->last_picture_ptr != NULL) {
