--- libavcodec/internal.h.orig	2019-08-05 22:52:21.000000000 +0200
+++ libavcodec/internal.h	2019-09-08 19:21:09.569572124 +0200
@@ -290,7 +290,10 @@
 {
     if(samples == AV_NOPTS_VALUE)
         return AV_NOPTS_VALUE;
-    return av_rescale_q(samples, (AVRational){ 1, avctx->sample_rate },
+    AVRational rat;
+    rat.num=1;
+    rat.den=avctx->sample_rate;
+    return av_rescale_q(samples, rat,
                         avctx->time_base);
 }
 
