--- libavformat/mpegtsenc.c.orig	2022-07-22 19:58:39.000000000 +0200
+++ libavformat/mpegtsenc.c	2022-10-26 11:55:54.000000000 +0200
@@ -1755,7 +1755,7 @@
             av_log(s, AV_LOG_ERROR, "H.264 bitstream malformed, "
                    "no startcode found, use the video bitstream filter 'h264_mp4toannexb' to fix it "
                    "('-bsf:v h264_mp4toannexb' option with ffmpeg)\n");
-            return AVERROR_INVALIDDATA;
+            // MEANX : Avidemux does not align the packet well, ignore this error    return AVERROR_INVALIDDATA;
         }
         av_log(s, AV_LOG_WARNING, "H.264 bitstream error, startcode missing, size %d", pkt->size);
         if (pkt->size)
