--- libavcodec/mpegvideo.c.old	2012-05-04 12:32:31.952749353 +0200
+++ libavcodec/mpegvideo.c	2012-05-04 12:33:08.236749317 +0200
@@ -1492,7 +1492,8 @@
  */
 void ff_print_debug_info(MpegEncContext *s, AVFrame *pict)
 {
-    if (s->avctx->hwaccel || !pict || !pict->mb_type)
+   // MEANX : THIS IS BORKING FRAME WHEN VDPAU IS USED AS HWACCEL IS NOT SET
+  // if (s->avctx->hwaccel || !pict || !pict->mb_type)
         return;
 
     if (s->avctx->debug & (FF_DEBUG_SKIP | FF_DEBUG_QP | FF_DEBUG_MB_TYPE)) {
