--- libavutil/log.c.orig	2026-08-04 00:14:48.000000000 +0200
+++ libavutil/log.c	2026-09-12 23:13:08.993533348 +0200
@@ -42,6 +42,7 @@
 #include "common.h"
 #include "internal.h"
 #include "log.h"
+#include "avassert.h" // MEANX
 #include "thread.h"
 #include "time.h"
 #include "time_internal.h"
@@ -126,7 +127,7 @@
 
 #endif
 static int use_color = -1;
-
+static AV_FATAL_HANDLER *fatalHandler=NULL; // MEANX
 #if defined(_WIN32) && HAVE_SETCONSOLETEXTATTRIBUTE && HAVE_GETSTDHANDLE
 static void win_console_puts(const char *str)
 {
@@ -525,3 +526,18 @@
     missing_feature_sample(0, avc, msg, argument_list);
     va_end(argument_list);
 }
+
+// MEANX
+void av_fatalHandler(const char *why, const char *filez, int lineno)
+{
+    if(!fatalHandler)
+        printf("[CRASH] %s, file %s, lineno=%d\n",why,filez,lineno);
+    else
+        (fatalHandler)(why,lineno,filez);
+}
+
+void av_setFatalHandler(AV_FATAL_HANDLER *func)
+{
+    fatalHandler=func;
+}
+//
