diff --git libavcodec/vdpau.h libavcodec/vdpau.h
index a42ca01..4b70be3 100644
--- libavcodec/vdpau.h
+++ libavcodec/vdpau.h
@@ -231,6 +231,7 @@ struct vdpau_render_state {
 
     int state; ///< Holds FF_VDPAU_STATE_* values.
 
+    int refCount; /// Used by how many client
 #if AV_HAVE_INCOMPATIBLE_LIBAV_ABI
     /** picture parameter information for all supported codecs */
     union AVVDPAUPictureInfo info;
