--- libavcodec/hevc_parser.c.old	2017-02-08 07:28:24.025576908 +0100
+++ libavcodec/hevc_parser.c	2017-02-08 07:35:03.939231159 +0100
@@ -27,6 +27,13 @@
 #include "h2645_parse.h"
 #include "parser.h"
 
+/* MEANX */
+HEVCSPS *ff_hevc_parser_get_sps(AVCodecParserContext *s);
+HEVCPPS *ff_hevc_parser_get_pps(AVCodecParserContext *s);
+HEVCVPS *ff_hevc_parser_get_vps(AVCodecParserContext *s);
+/* /MEANX */
+
+
 #define START_CODE 0x000001 ///< start_code_prefix_one_3bytes
 
 #define IS_IRAP_NAL(nal) (nal->type >= 16 && nal->type <= 23)
@@ -176,6 +183,62 @@
 
 #if ADVANCED_PARSER
 /**
+ \fn ff_hevc_parser_get_sps
+\brief meanx : extract sps from context
+*/
+HEVCSPS *ff_hevc_parser_get_sps(AVCodecParserContext *s)
+{
+      int i;
+      HEVCParserContext *ctx = s->priv_data;
+      if(!ctx) return NULL;
+      HEVCContext       *h   = &ctx->h;
+      if(!h) return NULL;
+      HEVCParamSets     *ps = &h->ps;
+      if(!ps) return NULL;
+
+     HEVCSPS *sps = ps->sps;
+     return sps;
+}
+/**
+ \fn ff_hevc_parser_get_sps
+\brief meanx : extract sps from context
+*/
+HEVCPPS *ff_hevc_parser_get_pps(AVCodecParserContext *s)
+{
+      int i;
+      HEVCParserContext *ctx = s->priv_data;
+      if(!ctx) return NULL;
+      HEVCContext       *h   = &ctx->h;
+      if(!h) return NULL;
+      HEVCParamSets     *ps = &h->ps;
+      if(!ps) return NULL;
+
+     HEVCPPS *pps = ps->pps;
+     return pps;
+}
+/**
+ \fn ff_hevc_parser_get_sps
+\brief meanx : extract sps from context
+*/
+HEVCVPS *ff_hevc_parser_get_vps(AVCodecParserContext *s)
+{
+      int i;
+      HEVCParserContext *ctx = s->priv_data;
+      if(!ctx) return NULL;
+      HEVCContext       *h   = &ctx->h;
+      if(!h) return NULL;
+      HEVCParamSets     *ps = &h->ps;
+      if(!ps) return NULL;
+
+     HEVCVPS *vps = ps->vps;
+     return vps;
+}
+// /MEANX
+
+/**
+
+
+/**
  * Parse NAL units of found picture and decode some basic information.
  *
  * @param s parser context.
