--- libavcodec/mpegvideo_enc.c.orig	2021-04-21 23:50:33.492163927 +0200
+++ libavcodec/mpegvideo_enc.c	2021-04-22 16:48:54.226003523 +0200
@@ -524,14 +524,14 @@
                "Warning vbv_delay will be set to 0xFFFF (=VBR) as the "
                "specified vbv buffer is too large for the given bitrate!\n");
     }
-
+#if 0 // MEANX
     if ((avctx->flags & AV_CODEC_FLAG_4MV) && s->codec_id != AV_CODEC_ID_MPEG4 &&
         s->codec_id != AV_CODEC_ID_H263 && s->codec_id != AV_CODEC_ID_H263P &&
         s->codec_id != AV_CODEC_ID_FLV1) {
         av_log(avctx, AV_LOG_ERROR, "4MV not supported by codec\n");
         return AVERROR(EINVAL);
     }
-
+#endif // MEANX
     if (s->obmc && avctx->mb_decision != FF_MB_DECISION_SIMPLE) {
         av_log(avctx, AV_LOG_ERROR,
                "OBMC is only supported with simple mb decision\n");
@@ -630,7 +630,7 @@
         s->mpeg_quant = 1;
     FF_ENABLE_DEPRECATION_WARNINGS
 #endif
-
+#if 0 // MEANX
     // FIXME mpeg2 uses that too
     if (s->mpeg_quant && (   s->codec_id != AV_CODEC_ID_MPEG4
                           && s->codec_id != AV_CODEC_ID_MPEG2VIDEO)) {
@@ -638,7 +638,7 @@
                "mpeg2 style quantization not supported by codec\n");
         return AVERROR(EINVAL);
     }
-
+#endif // MEANX
     if ((s->mpv_flags & FF_MPV_FLAG_CBP_RD) && !avctx->trellis) {
         av_log(avctx, AV_LOG_ERROR, "CBP RD needs trellis quant\n");
         return AVERROR(EINVAL);
