diff --git libavcodec/avcodec.h libavcodec/avcodec.h
index 4aeb57a..076721a 100644
--- libavcodec/avcodec.h
+++ libavcodec/avcodec.h
@@ -768,6 +768,10 @@ typedef struct RcOverride{
 #define CODEC_FLAG2_DROP_FRAME_TIMECODE 0x00002000 ///< timecode is in drop frame format. DEPRECATED!!!!
 #define CODEC_FLAG2_IGNORE_CROP   0x00010000 ///< Discard cropping information from SPS.
 
+//MEANX: NEVER EVER USE CLOSED GOP ?
+#define CODEC_FLAG2_32_PULLDOWN   0x80000000
+// /MEANX
+
 #define CODEC_FLAG2_CHUNKS        0x00008000 ///< Input bitstream might be truncated at a packet boundaries instead of only at frame boundaries.
 #define CODEC_FLAG2_SHOW_ALL      0x00400000 ///< Show all frames before the first keyframe
 #define CODEC_FLAG2_EXPORT_MVS    0x10000000 ///< Export motion vectors through frame side data
@@ -2302,6 +2306,7 @@ typedef struct AVCodecContext {
      * - decoding: unused
      */
     int rc_buffer_size;
+    int rc_buffer_size_header;  /*< That one is set in the header MEANX*/
 
     /**
      * ratecontrol override, see RcOverride
@@ -2325,6 +2330,7 @@ typedef struct AVCodecContext {
      * - decoding: Set by libavcodec.
      */
     int rc_max_rate;
+    int rc_max_rate_header; /*< That one is set in the header MEANX */
 
     /**
      * minimum bitrate
