--- /c/dev/adm_2.6_dev/avidemux_core/ffmpeg_package/patches/ffmpeg/libavcodec/mpegvideo.c	2012-04-18 15:14:07 +1000
+++ libavcodec/mpegvideo.c	2012-04-18 15:14:07 +1000
@@ -1566,8 +1566,10 @@
         }
     }
 
-    if ((s->avctx->debug & (FF_DEBUG_VIS_QP | FF_DEBUG_VIS_MB_TYPE)) ||
-        (s->avctx->debug_mv)) {
+	//MEANX: Allocate them always, as they may be free when we decode 1st image
+    /*if ((s->avctx->debug & (FF_DEBUG_VIS_QP | FF_DEBUG_VIS_MB_TYPE)) ||
+        (s->avctx->debug_mv)) {*/
+	{
         const int shift = 1 + s->quarter_sample;
         int mb_y;
         uint8_t *ptr;
