--- /e/dev/adm_2.6_dev/avidemux_core/ffmpeg_package/patches/ffmpeg/libavformat/isom.c	2012-05-29 18:18:34 +1000
+++ libavformat/isom.c	2012-05-29 18:18:34 +1000
@@ -257,7 +257,10 @@
     { CODEC_ID_MACE6,           MKTAG('M', 'A', 'C', '6') },
     { CODEC_ID_MP1,             MKTAG('.', 'm', 'p', '1') },
     { CODEC_ID_MP2,             MKTAG('.', 'm', 'p', '2') },
-    { CODEC_ID_MP3,             MKTAG('.', 'm', 'p', '3') },
+    //MEANX { CODEC_ID_MP3,             MKTAG('.', 'm', 'p', '3') },
+    { CODEC_ID_MP3, MKTAG( 'm', 'p', '4','a') },
+    { CODEC_ID_MP2, MKTAG( 'm', 'p', '4', 'a') },
+    // /MEANX
     { CODEC_ID_MP3,             0x6D730055                },
     { CODEC_ID_NELLYMOSER,      MKTAG('n', 'm', 'o', 's') }, /* Flash Media Server */
     { CODEC_ID_PCM_ALAW,        MKTAG('a', 'l', 'a', 'w') },
