--- libavcodec/avcodec.h.orig	2019-08-05 22:52:21.000000000 +0200
+++ libavcodec/avcodec.h	2019-09-08 19:21:09.545572268 +0200
@@ -945,6 +945,10 @@
  */
 #define AV_CODEC_FLAG2_IGNORE_CROP    (1 << 16)
 
+//MEANX: NEVER EVER USE CLOSED GOP ?
+#define AV_CODEC_FLAG2_32_PULLDOWN    (1 << 17)
+// /MEANX
+
 /**
  * Show all frames before the first keyframe
  */
@@ -2426,6 +2430,7 @@
      * - decoding: unused
      */
     int rc_buffer_size;
+    int rc_buffer_size_header; // MEANX : We fill the mpeg header with that value
 
     /**
      * ratecontrol override, see RcOverride
@@ -2441,6 +2446,7 @@
      * - decoding: Set by user, may be overwritten by libavcodec.
      */
     int64_t rc_max_rate;
+    int64_t rc_max_rate_header; // MEANX : We use that value to fill the mpeg header
 
     /**
      * minimum bitrate
