--- /e/dev/adm_2.6_dev/avidemux_core/ffmpeg_package/patches/ffmpeg/libavcodec/vdpau.h	2012-05-29 18:18:34 +1000
+++ libavcodec/vdpau.h	2012-05-29 18:18:34 +1000
@@ -72,7 +72,7 @@
     VdpVideoSurface surface; ///< Used as rendered surface, never changed.
 
     int state; ///< Holds FF_VDPAU_STATE_* values.
-
+    int refCount; /// Used by how many client
     /** Describe size/location of the compressed video data.
         Set to 0 when freeing bitstream_buffers. */
     int bitstream_buffers_allocated;
