--- libavcodec/internal.h.orig	2021-04-08 23:28:39.000000000 +0200
+++ libavcodec/internal.h	2021-04-21 23:48:01.932086286 +0200
@@ -274,7 +274,10 @@
 {
     if(samples == AV_NOPTS_VALUE)
         return AV_NOPTS_VALUE;
-    return av_rescale_q(samples, (AVRational){ 1, avctx->sample_rate },
+    AVRational rat;
+    rat.num=1;
+    rat.den=avctx->sample_rate;
+    return av_rescale_q(samples, rat,
                         avctx->time_base);
 }
 
