--- libavcodec/Makefile.org	2011-04-30 10:30:26.942213542 +0200
+++ libavcodec/Makefile	2011-04-30 10:30:47.412213539 +0200
@@ -539,7 +539,7 @@
                                           flacdec.o flacdata.o flac.o
 OBJS-$(CONFIG_MATROSKA_DEMUXER)        += mpeg4audio.o mpegaudiodata.o
 OBJS-$(CONFIG_MATROSKA_MUXER)          += xiph.o mpeg4audio.o \
-                                          flacdec.o flacdata.o flac.o \
+                                          flacdec.o flacdata.o flac.o vorbis_data.o \
                                           mpegaudiodata.o
 OBJS-$(CONFIG_MOV_DEMUXER)             += mpeg4audio.o mpegaudiodata.o
 OBJS-$(CONFIG_MOV_MUXER)               += mpeg4audio.o mpegaudiodata.o
