--- libavutil/hwcontext_dxva2.h.org	2017-03-18 16:33:48.003931867 +0100
+++ libavutil/hwcontext_dxva2.h	2017-03-18 16:34:03.859879370 +0100
@@ -33,6 +33,15 @@
 #include <d3d9.h>
 #include <dxva2api.h>
 
+
+typedef IDirect3D9       *(*ff_allocate_d3d)();
+typedef void             (*ff_release_d3d)(IDirect3D9 *);
+typedef IDirect3DDevice9 *(*ff_allocate_d3d_device)(IDirect3D9 *z);
+typedef void             (*ff_release_d3d_device)(IDirect3DDevice9 *);
+
+
+void ff_dxva2_set_handlers(ff_allocate_d3d allocateD3D, ff_release_d3d releaseD3D,ff_allocate_d3d_device allocateDevice,ff_release_d3d_device release_device  );
+
 /**
  * This struct is allocated as AVHWDeviceContext.hwctx
  */
