--- /e/dev/adm_2.6_dev/avidemux_core/ffmpeg_package/patches/ffmpeg/libavcodec/mpegvideo.c	2012-05-29 18:18:34 +1000
+++ libavcodec/mpegvideo.c	2012-05-29 18:18:34 +1000
@@ -1498,7 +1498,8 @@
  */
 void ff_print_debug_info(MpegEncContext *s, AVFrame *pict)
 {
-    if (s->avctx->hwaccel || !pict || !pict->mb_type)
+   // MEANX : THIS IS BORKING FRAME WHEN VDPAU IS USED AS HWACCEL IS NOT SET
+  // if (s->avctx->hwaccel || !pict || !pict->mb_type)
         return;
 
     if (s->avctx->debug & (FF_DEBUG_SKIP | FF_DEBUG_QP | FF_DEBUG_MB_TYPE)) {
@@ -1572,8 +1573,10 @@
         }
     }
 
-    if ((s->avctx->debug & (FF_DEBUG_VIS_QP | FF_DEBUG_VIS_MB_TYPE)) ||
-        (s->avctx->debug_mv)) {
+	//MEANX: Allocate them always, as they may be free when we decode 1st image
+    /*if ((s->avctx->debug & (FF_DEBUG_VIS_QP | FF_DEBUG_VIS_MB_TYPE)) ||
+        (s->avctx->debug_mv)) {*/
+	{
         const int shift = 1 + s->quarter_sample;
         int mb_y;
         uint8_t *ptr;
