--- libavcodec/x86/fmtconvert_mmx.c	2011-08-10 14:15:51.000000000 +0200
+++ libavcodec/x86/fmtconvert_mmx.c	2011-09-03 12:38:36.365349837 +0200
@@ -285,6 +285,7 @@
                 c->float_to_int16_interleave = float_to_int16_interleave_3dn2;
             }
         }
+#if 0 // MEANX : Disable as we cant guarantee output buffer is aligned
         if(mm_flags & AV_CPU_FLAG_SSE){
             c->int32_to_float_fmul_scalar = int32_to_float_fmul_scalar_sse;
             c->float_to_int16 = float_to_int16_sse;
@@ -298,5 +299,6 @@
             c->float_to_int16 = float_to_int16_sse2;
             c->float_to_int16_interleave = float_to_int16_interleave_sse2;
         }
+#endif
     }
 }
