--- libavformat/mpegenc.c.orig	2019-08-05 22:52:21.000000000 +0200
+++ libavformat/mpegenc.c	2019-09-08 19:44:28.824111846 +0200
@@ -984,7 +984,7 @@
                scr > pkt_desc->dts) { // FIXME: > vs >=
             if (stream->buffer_index < pkt_desc->size ||
                 stream->predecode_packet == stream->premux_packet) {
-                av_log(ctx, AV_LOG_ERROR,
+                av_log(ctx, AV_LOG_DEBUG, // EUM: don't flood the log during normal operation
                        "buffer underflow st=%d bufi=%d size=%d\n",
                        i, stream->buffer_index, pkt_desc->size);
                 break;
