--- libavformat/mpegtsenc.c.orig	2026-08-04 00:14:48.000000000 +0200
+++ libavformat/mpegtsenc.c	2026-09-12 23:13:08.986534644 +0200
@@ -1781,7 +1781,7 @@
             av_log(s, AV_LOG_ERROR, "%s bitstream malformed, "
                    "no startcode found, use the video bitstream filter '%s_mp4toannexb' to fix it "
                    "('-bsf:v %s_mp4toannexb' option with ffmpeg)\n", codec, codec, codec);
-            return AVERROR_INVALIDDATA;
+            // MEANX : Avidemux does not align the packet well, ignore this error    return AVERROR_INVALIDDATA;
         }
         av_log(s, AV_LOG_WARNING, "%s bitstream error, startcode missing, size %d", codec, pkt->size);
         if (pkt->size)
