--- libavcodec/internal.h.orig	2017-10-15 17:59:37.000000000 +0200
+++ libavcodec/internal.h	2017-10-20 21:57:43.593220641 +0200
@@ -282,7 +282,10 @@
 {
     if(samples == AV_NOPTS_VALUE)
         return AV_NOPTS_VALUE;
-    return av_rescale_q(samples, (AVRational){ 1, avctx->sample_rate },
+    AVRational rat;
+    rat.num=1;
+    rat.den=avctx->sample_rate;
+    return av_rescale_q(samples, rat,
                         avctx->time_base);
 }
 
