--- libavcodec/ratecontrol.c.orig	2022-07-22 19:58:39.000000000 +0200
+++ libavcodec/ratecontrol.c	2022-10-26 10:22:46.000000000 +0200
@@ -603,7 +603,11 @@
         }
 
         if (init_pass2(s) < 0) {
+            /* Calling ff_rate_control_uninit a second time results in memory corruption.
+             * Rely on avcodec_close and skip this one. (eum.) */
+#if 0
             ff_rate_control_uninit(s);
+#endif
             return -1;
         }
     }
