--- libavcodec/x86/fmtconvert_init.c.org	2012-10-08 09:29:46.977967202 +0200
+++ libavcodec/x86/fmtconvert_init.c	2012-10-08 09:30:31.697967791 +0200
@@ -132,6 +132,7 @@
                 c->float_to_int16_interleave = float_to_int16_interleave_3dnowext;
             }
         }
+#if 0 // MEANX : Cant guarantee proper alignment
         if (EXTERNAL_SSE(mm_flags)) {
             c->int32_to_float_fmul_scalar = ff_int32_to_float_fmul_scalar_sse;
             c->float_to_int16 = ff_float_to_int16_sse;
@@ -143,6 +144,7 @@
             c->float_to_int16 = ff_float_to_int16_sse2;
             c->float_to_int16_interleave = float_to_int16_interleave_sse2;
         }
+#endif // MEANX
     }
 #endif /* HAVE_YASM */
 }
