--- /c/dev/adm_2.6_dev/avidemux_core/ffmpeg_package/patches/ffmpeg/libavcodec/vdpau.h	2012-04-18 15:14:07 +1000
+++ libavcodec/vdpau.h	2012-04-18 15:14:07 +1000
@@ -66,7 +66,7 @@
     VdpVideoSurface surface; ///< Used as rendered surface, never changed.
 
     int state; ///< Holds FF_VDPAU_STATE_* values.
-
+    int refCount; /// Used by how many client
     /** Describe size/location of the compressed video data.
         Set to 0 when freeing bitstream_buffers. */
     int bitstream_buffers_allocated;
