--- libavutil/log.c.old	2016-02-16 07:29:29.585823781 +0100
+++ libavutil/log.c	2016-02-16 07:53:03.593024546 +0100
@@ -39,6 +39,7 @@
 #include "common.h"
 #include "internal.h"
 #include "log.h"
+#include "avassert.h" // MEANX
 
 #if HAVE_PTHREADS
 #include <pthread.h>
@@ -121,7 +122,7 @@
 
 #endif
 static int use_color = -1;
-
+static AV_FATAL_HANDLER *fatalHandler=NULL; // MEANX
 static void check_color_terminal(void)
 {
 #if defined(_WIN32) && !defined(__MINGW32CE__) && HAVE_SETCONSOLETEXTATTRIBUTE
@@ -430,7 +431,22 @@
     missing_feature_sample(0, avc, msg, argument_list);
     va_end(argument_list);
 }
+// MEANX
 
+void av_fatalHandler(const char *why,const char *filez, int lineno)
+{
+char *cleartext[2048];
+        snprintf((char *)cleartext,2000,"FFmpeg fatal error %s, in file %s at line %d\n",why,filez,lineno);
+        if(!fatalHandler)
+                printf("[CRASH] %s\n",cleartext);
+        else
+                (fatalHandler)(cleartext);
+}
+void av_setFatalHandler(AV_FATAL_HANDLER *func)
+{
+        fatalHandler=func;
+}
+//
 #ifdef TEST
 // LCOV_EXCL_START
 #include <string.h>
