--- libavformat/mpegtsenc.c.orig	2014-11-22 21:29:34.325333028 -0500
+++ libavformat/mpegtsenc.c	2014-11-22 21:37:43.354509274 -0500
@@ -1224,7 +1224,7 @@
             if (!st->nb_frames) {
                 av_log(s, AV_LOG_ERROR, "H.264 bitstream malformed, "
                        "no startcode found, use the h264_mp4toannexb bitstream filter (-bsf h264_mp4toannexb)\n");
-                return AVERROR(EINVAL);
+                // MEANX : Avidemux does not align the packet well, ignore this error    return AVERROR(EINVAL);
             }
             av_log(s, AV_LOG_WARNING, "H.264 bitstream error, startcode missing\n");
         }
