--- /home/fx/workspace/gitted/avidemux_core/ffmpeg_package/source/libavcodec/mpeg12.c	2011-08-10 14:18:20.000000000 +0200
+++ libavcodec/mpeg12.c	2011-09-03 08:35:49.555349178 +0200
@@ -1998,6 +1998,11 @@
         ff_er_frame_end(s);
 
         MPV_frame_end(s);
+/* MEANX */
+  if(s->current_picture_ptr)
+      s->current_picture_ptr->opaque=pict->opaque;
+/* MEANX */
+
 
         if (s->pict_type == AV_PICTURE_TYPE_B || s->low_delay) {
             *pict= *(AVFrame*)s->current_picture_ptr;
