--- libavcodec/libavcodec.v.old	2017-02-08 07:32:11.732241178 +0100
+++ libavcodec/libavcodec.v	2017-02-08 07:32:31.504125207 +0100
@@ -4,6 +4,12 @@
         #deprecated, remove after next bump
         audio_resample;
         audio_resample_close;
+ # Avidemux
+        ff_h264_info;
+        ff_golomb_vlc_len;
+        ff_ue_golomb_vlc_code;
+        ff_se_golomb_vlc_code;
+        ff_hevc_parser_get_*; 
     local:
         *;
 };
