diff --git libavcodec/golomb.h libavcodec/golomb.h
index d30bb6b..c19cee9 100644
--- libavcodec/golomb.h
+++ libavcodec/golomb.h
@@ -37,9 +37,16 @@
 
 #define INVALID_VLC           0x80000000
 
+#ifdef _MSC_VER
+extern __declspec(dllimport) const uint8_t ff_golomb_vlc_len[512];
+extern __declspec(dllimport) const uint8_t ff_ue_golomb_vlc_code[512];
+extern __declspec(dllimport) const  int8_t ff_se_golomb_vlc_code[512];
+#else
 extern const uint8_t ff_golomb_vlc_len[512];
 extern const uint8_t ff_ue_golomb_vlc_code[512];
 extern const  int8_t ff_se_golomb_vlc_code[512];
+#endif
+
 extern const uint8_t ff_ue_golomb_len[256];
 
 extern const uint8_t ff_interleaved_golomb_vlc_len[256];
