--- libavformat/matroskaenc.c.orig	2021-11-14 00:21:19.797422581 +0100
+++ libavformat/matroskaenc.c	2021-11-14 00:21:19.800422562 +0100
@@ -2797,7 +2797,7 @@
         if (pkt->size > 2 && (AV_RB16(pkt->data) & 0xfff0) == 0xfff0)
             ret = ff_stream_add_bitstream_filter(st, "aac_adtstoasc", NULL);
     } else if (st->codecpar->codec_id == AV_CODEC_ID_VP9) {
-        ret = ff_stream_add_bitstream_filter(st, "vp9_superframe", NULL);
+        // MEANX ret = ff_stream_add_bitstream_filter(st, "vp9_superframe", NULL);
     }
 
     return ret;
