--- /e/dev/adm_2.6_dev/avidemux_core/ffmpeg_package/patches/ffmpeg/libavformat/libavformat.v	2012-05-29 19:17:15 +1000
+++ libavformat/libavformat.v	2012-05-29 19:17:15 +1000
@@ -17,6 +17,7 @@
                 ffurl_seek;
                 ffurl_size;
                 ffurl_write;
+                ffurl_protocol_next;
                 url_open;
                 url_close;
                 url_write;
