--- libavformat/mpegtsenc.c.orig	2014-11-28 02:29:40.115456189 -0500
+++ libavformat/mpegtsenc.c		2014-11-28 02:39:25.579440572 -0500
@@ -1195,7 +1195,7 @@
             av_log(s, AV_LOG_ERROR, "H.264 bitstream malformed, "
                    "no startcode found, use the video bitstream filter 'h264_mp4toannexb' to fix it "
                    "('-bsf:v h264_mp4toannexb' option with ffmpeg)\n");
-            return AVERROR_INVALIDDATA;
+            // MEANX : Avidemux does not align the packet well, ignore this error    return AVERROR_INVALIDDATA;
         }
         av_log(s, AV_LOG_WARNING, "H.264 bitstream error, startcode missing\n");
     }
