--- libavcodec/encode.h.orig	2026-08-04 00:14:47.000000000 +0200
+++ libavcodec/encode.h	2026-09-12 22:39:12.931581119 +0200
@@ -93,9 +93,12 @@
 static av_always_inline int64_t ff_samples_to_time_base(const AVCodecContext *avctx,
                                                         int64_t samples)
 {
+    AVRational rat;
     if (samples == AV_NOPTS_VALUE)
         return AV_NOPTS_VALUE;
-    return av_rescale_q(samples, (AVRational){ 1, avctx->sample_rate },
+    rat.num = 1;
+    rat.den = avctx->sample_rate;
+    return av_rescale_q(samples, rat,
                         avctx->time_base);
 }
 
