--- libavutil/log.c.orig	2022-07-22 19:58:39.000000000 +0200
+++ libavutil/log.c	2022-10-26 11:55:54.000000000 +0200
@@ -41,6 +41,7 @@
 #include "common.h"
 #include "internal.h"
 #include "log.h"
+#include "avassert.h" // MEANX
 #include "thread.h"
 
 static AVMutex mutex = AV_MUTEX_INITIALIZER;
@@ -121,7 +122,7 @@
 
 #endif
 static int use_color = -1;
-
+static AV_FATAL_HANDLER *fatalHandler=NULL; // MEANX
 #if defined(_WIN32) && HAVE_SETCONSOLETEXTATTRIBUTE && HAVE_GETSTDHANDLE
 static void win_console_puts(const char *str)
 {
@@ -490,3 +491,18 @@
     missing_feature_sample(0, avc, msg, argument_list);
     va_end(argument_list);
 }
+
+// MEANX
+void av_fatalHandler(const char *why, const char *filez, int lineno)
+{
+    if(!fatalHandler)
+        printf("[CRASH] %s, file %s, lineno=%d\n",why,filez,lineno);
+    else
+        (fatalHandler)(why,lineno,filez);
+}
+
+void av_setFatalHandler(AV_FATAL_HANDLER *func)
+{
+    fatalHandler=func;
+}
+//
