--- libavcodec/x86/fmtconvert_init.c.orig	2014-11-01 08:52:42.000000000 -0400
+++ libavcodec/x86/fmtconvert_init.c	2014-11-22 21:15:46.964170617 -0500
@@ -132,6 +132,7 @@
             c->float_to_int16_interleave = float_to_int16_interleave_3dnowext;
         }
     }
+#if 0 // MEANX : Cant guarantee proper alignment
     if (EXTERNAL_SSE(cpu_flags)) {
         c->int32_to_float_fmul_scalar = ff_int32_to_float_fmul_scalar_sse;
         c->float_to_int16             = ff_float_to_int16_sse;
@@ -143,5 +144,6 @@
         c->float_to_int16             = ff_float_to_int16_sse2;
         c->float_to_int16_interleave  = float_to_int16_interleave_sse2;
     }
+#endif // MEANX
 #endif /* HAVE_YASM */
 }
