--- libavformat/matroskaenc.c.orig	2021-04-22 19:59:01.050122387 +0200
+++ libavformat/matroskaenc.c	2021-04-22 20:01:07.674346845 +0200
@@ -2796,7 +2796,7 @@
         if (pkt->size > 2 && (AV_RB16(pkt->data) & 0xfff0) == 0xfff0)
             ret = ff_stream_add_bitstream_filter(st, "aac_adtstoasc", NULL);
     } else if (st->codecpar->codec_id == AV_CODEC_ID_VP9) {
-        ret = ff_stream_add_bitstream_filter(st, "vp9_superframe", NULL);
+        // MEANX ret = ff_stream_add_bitstream_filter(st, "vp9_superframe", NULL);
     }
 
     return ret;
