--- libavcodec/libavcodec.v.orig	2014-11-01 21:47:34.000000000 -0400
+++ libavcodec/libavcodec.v		2014-11-28 03:28:34.500032881 -0500
@@ -3,5 +3,10 @@
                 #deprecated, remove after next bump
                 audio_resample;
                 audio_resample_close;
+                # Avidemux
+                ff_h264_info;
+                ff_golomb_vlc_len;
+                ff_ue_golomb_vlc_code;
+                ff_se_golomb_vlc_code;
         local:  *;
 };
