--- libavformat/mpegtsenc.c	2012-01-26 23:15:59.000000000 +0100
+++ libavformat/mpegtsenc.c	2012-05-01 18:13:15.990894403 +0200
@@ -996,8 +996,8 @@
 
         if (pkt->size < 5 || AV_RB32(pkt->data) != 0x0000001) {
             av_log(s, AV_LOG_ERROR, "H.264 bitstream malformed, "
-                   "no startcode found, use the h264_mp4toannexb bitstream filter\n");
-            return AVERROR_INVALIDDATA;
+                   "no startcode found, use the h264_mp4toannexb bitstream filter ( size=%d %02x:%02x:%02x:%02x:%02x\n )",pkt->size,pkt->data[0],pkt->data[1],pkt->data[2],pkt->data[3],pkt->data[4]);
+            //return AVERROR_INVALIDDATA; MEANX Avidemux does not align well the packet + does not work with short packet
         }
 
         do {
