--- libavcodec/golomb.h.orig	2017-10-20 21:08:20.663723256 +0200
+++ libavcodec/golomb.h	2017-10-20 21:07:29.184053505 +0200
@@ -32,14 +32,22 @@
 
 #include <stdint.h>
 
+#include "libavutil/internal.h"
 #include "get_bits.h"
 #include "put_bits.h"
 
 #define INVALID_VLC           0x80000000
 
+#ifdef _MSC_VER
+extern __declspec(dllimport) const uint8_t ff_golomb_vlc_len[512];
+extern __declspec(dllimport) const uint8_t ff_ue_golomb_vlc_code[512];
+extern __declspec(dllimport) const  int8_t ff_se_golomb_vlc_code[512];
+#else
 extern const uint8_t ff_golomb_vlc_len[512];
 extern const uint8_t ff_ue_golomb_vlc_code[512];
 extern const  int8_t ff_se_golomb_vlc_code[512];
+#endif
+
 extern const uint8_t ff_ue_golomb_len[256];
 
 extern const uint8_t ff_interleaved_golomb_vlc_len[256];
