diff --git libavcodec/mpegvideo_enc.c libavcodec/mpegvideo_enc.c
index 49fc083..a6ed8da 100644
--- libavcodec/mpegvideo_enc.c
+++ libavcodec/mpegvideo_enc.c
@@ -469,14 +469,14 @@ av_cold int ff_mpv_encode_init(AVCodecContext *avctx)
                "Warning vbv_delay will be set to 0xFFFF (=VBR) as the "
                "specified vbv buffer is too large for the given bitrate!\n");
     }
-
+#if 0 // MEANX
     if ((s->avctx->flags & CODEC_FLAG_4MV) && s->codec_id != AV_CODEC_ID_MPEG4 &&
         s->codec_id != AV_CODEC_ID_H263 && s->codec_id != AV_CODEC_ID_H263P &&
         s->codec_id != AV_CODEC_ID_FLV1) {
         av_log(avctx, AV_LOG_ERROR, "4MV not supported by codec\n");
         return -1;
     }
-
+#endif // MEANX
     if (s->obmc && s->avctx->mb_decision != FF_MB_DECISION_SIMPLE) {
         av_log(avctx, AV_LOG_ERROR,
                "OBMC is only supported with simple mb decision\n");
@@ -568,7 +568,7 @@ av_cold int ff_mpv_encode_init(AVCodecContext *avctx)
         av_log(avctx, AV_LOG_ERROR, "interlacing not supported by codec\n");
         return -1;
     }
-
+#if 0 // MEANX
     // FIXME mpeg2 uses that too
     if (s->mpeg_quant && (   s->codec_id != AV_CODEC_ID_MPEG4
                           && s->codec_id != AV_CODEC_ID_MPEG2VIDEO)) {
@@ -576,7 +576,7 @@ av_cold int ff_mpv_encode_init(AVCodecContext *avctx)
                "mpeg2 style quantization not supported by codec\n");
         return -1;
     }
-
+#endif
     if ((s->mpv_flags & FF_MPV_FLAG_CBP_RD) && !avctx->trellis) {
         av_log(avctx, AV_LOG_ERROR, "CBP RD needs trellis quant\n");
         return -1;
