--- libavcodec/vdpau.h	2011-08-10 14:18:20.000000000 +0200
+++ libavcodec/vdpau.h	2011-09-03 08:35:49.735349178 +0200
@@ -66,7 +66,7 @@
     VdpVideoSurface surface; ///< Used as rendered surface, never changed.
 
     int state; ///< Holds FF_VDPAU_STATE_* values.
-
+    int refCount; /// Used by how many client
     /** Describe size/location of the compressed video data.
         Set to 0 when freeing bitstream_buffers. */
     int bitstream_buffers_allocated;
