diff --git a/avidemux_core/ADM_ffmpeg/libswscale/swscale_template.c b/avidemux_core/ADM_ffmpeg/libswscale/swscale_template.c
index 34868ac..88575e6 100644
--- a/avidemux_core/ADM_ffmpeg/libswscale/swscale_template.c
+++ b/avidemux_core/ADM_ffmpeg/libswscale/swscale_template.c
@@ -951,7 +951,7 @@ static inline void RENAME(yuv2yuv1)(SwsContext *c, const int16_t *lumSrc, const
 #if COMPILE_TEMPLATE_MMX
     if(!(c->flags & SWS_BITEXACT)) {
         long p= 4;
-        const uint8_t *src[4]= {alpSrc + dstW, lumSrc + dstW, chrSrc + chrDstW, chrSrc + VOFW + chrDstW};
+        const uint8_t *src[4]= {alpSrc + dstW, (uint8_t *)lumSrc + dstW, (uint8_t *)chrSrc + chrDstW, (uint8_t *)chrSrc + VOFW + chrDstW}; // MEANX
         uint8_t *dst[4]= {aDest, dest, uDest, vDest};
         x86_reg counter[4]= {dstW, dstW, chrDstW, chrDstW};
 
