--- mozilla/nsprpub/configure.orig	2012-02-15 06:06:00 +1100
+++ mozilla/nsprpub/configure	2012-02-21 14:07:22 +1100
@@ -748,7 +748,7 @@
 USE_IPV6=
 USE_MDUPDATE=
 _MACOSX_DEPLOYMENT_TARGET=
-_OPTIMIZE_FLAGS=-O
+_OPTIMIZE_FLAGS=-O3
 _DEBUG_FLAGS=-g
 MOZ_DEBUG=1
 MOZ_OPTIMIZE=
@@ -4536,10 +4536,10 @@
         CC="$CC -mwindows"
         CXX="$CXX -mwindows"
         DLL_SUFFIX=dll
-        MKSHLIB='$(CC) -shared -Wl,--export-all-symbols -Wl,--out-implib -Wl,$(IMPORT_LIBRARY) $(DLLBASE) -o $(subst $(OBJDIR)/,,$(SHARED_LIBRARY))'
+        MKSHLIB='$(CC) $(CFLAGS) -shared -Wl,--export-all-symbols -Wl,--out-implib -Wl,$(IMPORT_LIBRARY) $(DLLBASE) -o $(subst $(OBJDIR)/,,$(SHARED_LIBRARY))'
         RC=$WINDRES
         # Use temp file for windres (bug 213281)
-        RCFLAGS='-O coff --use-temp-file'
+        RCFLAGS='-F pe-i386 -O coff --use-temp-file'
     else
         CC=cl
         CXX=cl
