diff --git a/avidemux/ADM_libraries/ADM_ffmpeg/libavutil/internal.h b/avidemux/ADM_libraries/ADM_ffmpeg/libavutil/internal.h
index 7a38fb1..bf845f8 100644
--- libavutil/internal.h
+++ libavutil/internal.h
@@ -34,7 +34,11 @@
 #include <stdint.h>
 #include <stddef.h>
 #include <assert.h>
+// MEANX
+#ifndef ADM_NO_CONFIG_H
 #include "config.h"
+#endif
+// MEANX
 #include "common.h"
 #include "mem.h"
 #include "timer.h"
