*** libavcodec/utils.c.old	Wed Apr  1 18:35:40 2009
--- libavcodec/utils.c	Wed Apr  1 18:35:40 2009
***************
*** 555,564 ****
--- 555,566 ----
  
      if((avctx->codec->capabilities & CODEC_CAP_DELAY) || buf_size){
          //FIXME remove the check below _after_ ensuring that all audio check that the available space is enough
+ #if 0 // MEANX : Silence
          if(*frame_size_ptr < AVCODEC_MAX_AUDIO_FRAME_SIZE){
              av_log(avctx, AV_LOG_ERROR, "buffer smaller than AVCODEC_MAX_AUDIO_FRAME_SIZE\n");
              return -1;
          }
+ #endif
          if(*frame_size_ptr < FF_MIN_BUFFER_SIZE ||
          *frame_size_ptr < avctx->channels * avctx->frame_size * sizeof(int16_t)){
              av_log(avctx, AV_LOG_ERROR, "buffer %d too small\n", *frame_size_ptr);
***************
*** 969,975 ****
          return -1;
      }
  #if !HAVE_MKSTEMP
!     fd = open(*filename, O_RDWR | O_BINARY | O_CREAT, 0444);
  #else
      snprintf(*filename, len, "/tmp/%sXXXXXX", prefix);
      fd = mkstemp(*filename);
--- 971,977 ----
          return -1;
      }
  #if !HAVE_MKSTEMP
!     fd = open(*filename, O_RDWR /* MEANX | O_BINARY*/ | O_CREAT, 0444);
  #else
      snprintf(*filename, len, "/tmp/%sXXXXXX", prefix);
      fd = mkstemp(*filename);
