*** libavutil/internal.h.old	Sat Feb 21 16:11:15 2009
--- libavutil/internal.h	Sat Feb 21 16:11:15 2009
***************
*** 215,226 ****
--- 215,228 ----
  #endif
  
  /* avoid usage of dangerous/inappropriate system functions */
+ #if 0 //MEANX
  #undef  malloc
  #define malloc please_use_av_malloc
  #undef  free
  #define free please_use_av_free
  #undef  realloc
  #define realloc please_use_av_realloc
+ #endif
  #undef  time
  #define time time_is_forbidden_due_to_security_issues
  #undef  rand
