Index: libavfilter/Makefile
===================================================================
--- libavfilter/Makefile	(revision 18300)
+++ libavfilter/Makefile	(working copy)
@@ -1,8 +1,8 @@
 include $(SUBDIR)../config.mak
 
-NAME = avfilter
-FFLIBS = avcodec avutil swscale
-FFLIBS-$(CONFIG_AVFILTER_LAVF) += avformat
+NAME = ADMavfilter
+FFLIBS = ADMavcodec ADMavutil ADMswscale
+FFLIBS-$(CONFIG_AVFILTER_LAVF) += ADMavformat
 
 HEADERS = avfilter.h
 
