Index: libavutil/Makefile
===================================================================
--- libavutil/Makefile	(revision 18300)
+++ libavutil/Makefile	(working copy)
@@ -1,6 +1,6 @@
 include $(SUBDIR)../config.mak
 
-NAME = avutil
+NAME = ADMavutil
 
 HEADERS = adler32.h                                                     \
           avstring.h                                                    \
