*** libavcodec/mpeg12.c.old	Wed Apr  1 18:35:38 2009
--- libavcodec/mpeg12.c	Wed Apr  1 18:35:38 2009
***************
*** 1952,1957 ****
--- 1952,1962 ----
          ff_er_frame_end(s);
  
          MPV_frame_end(s);
+ /* MEANX */
+   if(s->current_picture_ptr)
+       s->current_picture_ptr->opaque=pict->opaque;
+ /* MEANX */
+ 
  
          if (s->pict_type == FF_B_TYPE || s->low_delay) {
              *pict= *(AVFrame*)s->current_picture_ptr;
