diff --git a/avidemux/ADM_libraries/ADM_ffmpeg/libavcodec/x86/mathops.h b/avidemux/ADM_libraries/ADM_ffmpeg/libavcodec/x86/mathops.h
index a66c601..d8257f0 100644
--- libavcodec/x86/mathops.h
+++ libavcodec/x86/mathops.h
@@ -22,7 +22,12 @@
 #ifndef AVCODEC_X86_MATHOPS_H
 #define AVCODEC_X86_MATHOPS_H
 
+// MEANX
+#ifndef ADM_NO_CONFIG_H
 #include "config.h"
+#endif
+// MEANX
+
 #include "libavutil/common.h"
 
 #if ARCH_X86_32
