diff --git a/avidemux/ADM_libraries/ADM_ffmpeg/libavutil/x86/bswap.h b/avidemux/ADM_libraries/ADM_ffmpeg/libavutil/x86/bswap.h
index 3eeb5a4..82626f2 100644
--- libavutil/x86/bswap.h
+++ libavutil/x86/bswap.h
@@ -25,7 +25,12 @@
 #define AVUTIL_X86_BSWAP_H
 
 #include <stdint.h>
+// MEANX
+#ifndef ADM_NO_CONFIG_H
 #include "config.h"
+#endif
+// MEANX
+
 #include "libavutil/common.h"
 
 #define bswap_16 bswap_16
