diff --git a/src/qt/qt6/qt6-qtimageformats.mk b/src/qt/qt6/qt6-qtimageformats.mk
index ac7b4329..64403663 100644
--- a/src/qt/qt6/qt6-qtimageformats.mk
+++ b/src/qt/qt6/qt6-qtimageformats.mk
@@ -6,7 +6,7 @@ PKG := qt6-qtimageformats
 $(eval $(QT6_METADATA))
 
 $(PKG)_CHECKSUM := cecd8900f34b6550076309bc94f62f828008b633a4239e0a08c86788f41001f8
-$(PKG)_DEPS     := cc qt6-conf qt6-qtbase jasper libmng libwebp tiff
+$(PKG)_DEPS     := cc qt6-conf qt6-qtbase jasper libdeflate libmng libwebp tiff
 
 QT6_PREFIX   = '$(PREFIX)/$(TARGET)/$(MXE_QT6_ID)'
 QT6_QT_CMAKE = '$(QT6_PREFIX)/$(if $(findstring mingw,$(TARGET)),bin,libexec)/qt-cmake-private' \
