diff --git a/avidemux_core/ADM_ffmpeg/libavcodec/mpegvideo.c b/avidemux_core/ADM_ffmpeg/libavcodec/mpegvideo.c
index 684bdfe..2135c3d 100644
--- a/avidemux_core/ADM_ffmpeg/libavcodec/mpegvideo.c
+++ b/avidemux_core/ADM_ffmpeg/libavcodec/mpegvideo.c
@@ -652,7 +652,11 @@ av_cold int MPV_common_init(MpegEncContext *s)
     FF_ALLOCZ_OR_GOTO(s->avctx, s->prev_pict_types, PREV_PICT_TYPES_BUFFER_SIZE, fail);
 
     s->parse_context.state= -1;
-    if((s->avctx->debug&(FF_DEBUG_VIS_QP|FF_DEBUG_VIS_MB_TYPE)) || (s->avctx->debug_mv)){
+ //MEANX: Allocate them always, as they may be free when we decode 1st image
+
+    // MEANXif((s->avctx->debug&(FF_DEBUG_VIS_QP|FF_DEBUG_VIS_MB_TYPE)) || (s->avctx->debug_mv)){
+if(1){ // MEANX
+
        s->visualization_buffer[0] = av_malloc((s->mb_width*16 + 2*EDGE_WIDTH) * s->mb_height*16 + 2*EDGE_WIDTH);
        s->visualization_buffer[1] = av_malloc((s->mb_width*16 + 2*EDGE_WIDTH) * s->mb_height*16 + 2*EDGE_WIDTH);
        s->visualization_buffer[2] = av_malloc((s->mb_width*16 + 2*EDGE_WIDTH) * s->mb_height*16 + 2*EDGE_WIDTH);
