diff --git a/avidemux_core/ADM_ffmpeg/libavcodec/pngdec.c b/avidemux_core/ADM_ffmpeg/libavcodec/pngdec.c
index eb8989f..5ecdd56 100644
--- a/avidemux_core/ADM_ffmpeg/libavcodec/pngdec.c
+++ b/avidemux_core/ADM_ffmpeg/libavcodec/pngdec.c
@@ -21,7 +21,7 @@
 #include "libavcore/imgutils.h"
 #include "avcodec.h"
 #include "bytestream.h"
-#include "png.h"
+#include "ffpng.h"
 #include "dsputil.h"
 
 /* TODO:
