diff --git a/avidemux_core/ADM_ffmpeg/libavutil/intreadwrite.h b/avidemux_core/ADM_ffmpeg/libavutil/intreadwrite.h
index c8026f0..272a3b1 100644
--- a/avidemux_core/ADM_ffmpeg/libavutil/intreadwrite.h
+++ b/avidemux_core/ADM_ffmpeg/libavutil/intreadwrite.h
@@ -20,7 +20,12 @@
 #define AVUTIL_INTREADWRITE_H
 
 #include <stdint.h>
+// MEANX
+#ifndef ADM_NO_CONFIG_H
 #include "config.h"
+#endif
+// MEANX
+
 #include "bswap.h"
 #include "common.h"
 
