diff --git a/avidemux_core/ADM_ffmpeg/libavcodec/h264.c b/avidemux_core/ADM_ffmpeg/libavcodec/h264.c
index e4654f0..cfab425 100644
--- a/avidemux_core/ADM_ffmpeg/libavcodec/h264.c
+++ b/avidemux_core/ADM_ffmpeg/libavcodec/h264.c
@@ -3358,6 +3358,17 @@ av_cold int ff_h264_decode_end(AVCodecContext *avctx)
     return 0;
 }
 
+/* MEANX */
+int av_getAVCStreamInfo(AVCodecContext *avctx, uint32_t  *nalSize, uint32_t *isAvc)
+{
+      H264Context *h = avctx->priv_data;
+      assert(h);
+      *nalSize=h->nal_length_size;
+      *isAvc=h->is_avc;
+	return 1;
+
+  }
+/* MeanX */
 
 AVCodec h264_decoder = {
     "h264",
