diff --git a/avidemux_core/ADM_ffmpeg/libavcodec/fft.c b/avidemux_core/ADM_ffmpeg/libavcodec/fft.c
index bf240bf..6199d91 100644
--- a/avidemux_core/ADM_ffmpeg/libavcodec/fft.c
+++ b/avidemux_core/ADM_ffmpeg/libavcodec/fft.c
@@ -112,7 +112,7 @@ av_cold int ff_fft_init(FFTContext *s, int nbits, int inverse)
 
     if (ARCH_ARM)     ff_fft_init_arm(s);
     if (HAVE_ALTIVEC) ff_fft_init_altivec(s);
-    if (HAVE_MMX)     ff_fft_init_mmx(s);
+    // MEANX if (HAVE_MMX)     ff_fft_init_mmx(s);
 
     if (s->split_radix) {
         for(j=4; j<=nbits; j++) {
