diff --git a/avidemux_core/ADM_ffmpeg/libavcodec/ffv1.c b/avidemux_core/ADM_ffmpeg/libavcodec/ffv1.c
index 12056e1..b2f3efe 100644
--- a/avidemux_core/ADM_ffmpeg/libavcodec/ffv1.c
+++ b/avidemux_core/ADM_ffmpeg/libavcodec/ffv1.c
@@ -1107,6 +1107,8 @@ static int decode_frame(AVCodecContext *avctx, void *data, int *data_size, AVPac
         clear_state(f);
     }else{
         p->key_frame= 0;
+         p->pict_type= FF_P_TYPE; // MEANX : looks more like a P to me as user
+
     }
     if(f->ac>1){
         int i;
