diff --git a/avidemux_core/ADM_ffmpeg/libavformat/utils.c b/avidemux_core/ADM_ffmpeg/libavformat/utils.c
index 22c833d..e90b609 100644
--- a/avidemux_core/ADM_ffmpeg/libavformat/utils.c
+++ b/avidemux_core/ADM_ffmpeg/libavformat/utils.c
@@ -49,13 +49,14 @@ unsigned avformat_version(void)
 
 const char *avformat_configuration(void)
 {
-    return FFMPEG_CONFIGURATION;
+    return "avidemux"; //FFMPEG_CONFIGURATION;
 }
 
 const char *avformat_license(void)
 {
 #define LICENSE_PREFIX "libavformat license: "
-    return LICENSE_PREFIX FFMPEG_LICENSE + sizeof(LICENSE_PREFIX) - 1;
+    //return LICENSE_PREFIX FFMPEG_LICENSE + sizeof(LICENSE_PREFIX) - 1;
+        return LICENSE_PREFIX "GPL";
 }
 
 /* fraction handling */
