diff --git a/avidemux_core/ADM_ffmpeg/libavutil/avutil.h b/avidemux_core/ADM_ffmpeg/libavutil/avutil.h
index 50b29fc..691c5cb 100644
--- a/avidemux_core/ADM_ffmpeg/libavutil/avutil.h
+++ b/avidemux_core/ADM_ffmpeg/libavutil/avutil.h
@@ -26,6 +26,13 @@
  * external API header
  */
 
+/* MEANX
+ * - */
+#define ASMALIGN(ZEROBITS) ".p2align " #ZEROBITS "\n\t"
+/* /MEANX
+ */
+
+
 
 #define AV_STRINGIFY(s)         AV_TOSTRING(s)
 #define AV_TOSTRING(s) #s
